src/components/Header/index.tsx

```diff
@@ -245,8 +248,27 @@ const StyledExternalLink = styled(ExternalLink).attrs({
   }
 `
 
+function getCurrentPageFromLocation(locationPathname: string): PageName {
+  switch (locationPathname) {
+    case '/swap':
+      return PageName.SWAP_PAGE
+    case '/vote':
+      return PageName.VOTE_PAGE
+    case '/pool':
+      return PageName.POOL_PAGE
+    default:
+      return PageName.SWAP_PAGE
+  }
+}
+
 export default function Header() {
   const { account, chainId } = useWeb3React()
+  const location = useLocation()
+  const [currentPage, setCurrentPage] = useState(getCurrentPageFromLocation(location.pathname))
```

Comment:
won't `useLocation` re-render? so couldn't you just do `currentPage = getCurrentPageFromLocation(location.pathname)`

Reply:
yes you're totally right. done.